WireAPI/Sources/WireAPI/Network/PushChannel/PushChannel.swift

```diff
@@ -17,19 +17,141 @@
 //
 
 import Foundation
+import WireUtilitiesPkg
 
-// sourcery: AutoMockable
-/// Make a direct connection to a server to receive update events.
-public protocol PushChannelProtocol {
+final class PushChannel: NSObject, PushChannelProtocol {
 
-    /// Open the push channel and start receiving update events.
-    ///
-    /// - Returns: An async stream of live update event envelopes.
+    private let request: URLRequest
+    private var webSocket: (any WebSocketProtocol)?
+    private var webSocketProvider: (any WebSocketProvider)!
 
-    func open() async throws -> AsyncStream<UpdateEventEnvelope>
+    init(
+        request: URLRequest,
+        minTLSVersion: TLSVersion
+    ) {
+        self.request = request
+        super.init()
+        let factory = URLSessionConfigurationFactory(minTLSVersion: minTLSVersion)
```

Comment:
Perhaps this can be done by the caller of init. The way we only need a single init method